File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def dfs(cur, bef):
    global n
    for nxt in range(n):
        if nxt == bef:
            continue
        if computers[cur][nxt] and not v[nxt]:
            v[nxt] = 1
            dfs(nxt, cur)

def solution(n, computers):
    global n
    answer = 0
    v = [0] * n
    for i in range(n):
        if not v[i]:
            v[i] = 1
            dfs(i, -1)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int n, int[,] computers) {
        int answer = 0;
        bool[] v = new bool[n];
        for (int i=0; i<n; ++i) {
            if (!v[i]) {
                answer++;
                v[i] = true;
                var q = new Queue<int>();
                q.Enqueue(i);
                while (q.Count>0) {
                    int cur = q.Dequeue();
                    for (int nxt=0; nxt<n; ++nxt) {
                        if (computers[cur,nxt] == 1 && !v[nxt]) {
                            v[nxt] = true;
                            q.Enqueue(nxt);
                        }
                    }
                    
                }
            }
        }
        
        
        
        return answer;
    }
}